lib/promise.mli

```diff
@@ -0,0 +1,13 @@
+type 'a promise
+val _Promise : 'a
+val _catch : 'a promise Js.t -> ('b -> 'a promise Js.t) -> 'a promise Js.t
```

Comment:
`'b` is the rejection type, and can be any JS value. However, once we pass OCaml callbacks to JS `catch` and `then`, what will happen to OCaml exceptions raised in the callbacks? If they turn into JS exceptions, the JS promise functions will turn them into rejections, which will further complicate what `'b` can be, and we will need to prepare users of this promise API for that.